backend/logger/logger.py

```diff
@@ -0,0 +1,84 @@
+from file_handler import FileHandler
+from singleton import Singleton
+
+from datetime import datetime
+import enum
+import os
+from inspect import stack, getframeinfo
+
+
+class Level(enum.IntEnum):
+    DEBUG = 0
+    WARNING = 1
+    ERROR = 2
+
+
+class Logger(metaclass=Singleton):
+    """
+    Main class for logging functionalities.
+    """
+
+    def __init__(self, file_name="session", level=Level.DEBUG):
+        self.level = level
+        file_handler = FileHandler(file_name)
+        self.writer = file_handler.get_write_handle()
```

Comment:
Just wondering why we have a writer as a separate entity rather than just doing file_handler.writer here? We may want to reduce coupling here, for example we could make a function on the file_handler saying `logToFile` and then we can call that instead of going into the file_handler and pulling out the write handle

Reply:
It's because the FileHandler is detached from its writer. You need to query the writer from the file_handler. The file_handler itself is supposed to be an abstraction between the logger and the OS file system